.travis.yml

```diff
@@ -1,60 +1,74 @@
-sudo: required
-dist: trusty
+dist: xenial
 compiler:
   - gcc
   - clang
 language: c
-before_install:
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq --no-install-recommends
-      iptables-dev
-      libatasmart-dev
-      libcap-dev
-      libcurl4-gnutls-dev
-      libdbi0-dev
-      libesmtp-dev
-      libganglia1-dev
-      libgcrypt11-dev
-      libglib2.0-dev
-      libgps-dev
-      libhiredis-dev
-      libi2c-dev
-      libldap2-dev
-      libltdl-dev
-      liblua5.2-dev
-      liblvm2-dev
-      libmemcached-dev
-      libmicrohttpd-dev
-      libmnl-dev
-      libmodbus-dev
-      libmosquitto0-dev
-      libmysqlclient-dev
-      libnotify-dev
-      libopenipmi-dev
-      liboping-dev
-      libow-dev
-      libpcap-dev
-      libperl-dev
-      libpq-dev
-      libprotobuf-c0-dev
-      librabbitmq-dev
-      librdkafka-dev
-      librrd-dev
-      libsensors4-dev
-      libsigrok-dev
-      libsnmp-dev
-      libstatgrab-dev
-      libtokyocabinet-dev
-      libtokyotyrant-dev
-      libudev-dev
-      libupsclient-dev
-      libvarnish-dev
-      libvirt-dev
-      libxml2-dev
-      libyajl-dev
-      linux-libc-dev
-      perl
-      protobuf-c-compiler
-      python-dev
-# VERBOSE=1 dumps the contents of the test suites logs to standard output.
-script: sh build.sh && ./configure --with-libmongoc=own && make distcheck VERBOSE=1
+
+before_script: autoreconf -vif
```

Comment:
I wonder if we should instead run `sh clean.sh` as part of the script...

Reply:
Is that related to this change? We weren't running it before and I don't know what it does. I can review a follow-up change if you want to look into it.

Reply:
Yes, it's related. We weren't running `autoreconf` before either, and, unlike upstream, we don't run it as part of our normal package builds, while we do run `clean.sh`: [here](https://github.com/Stackdriver/agent-packaging/blob/stackdriver-agent-5.5.2/rpm/stackdriver-agent.spec#L270) and [here](https://github.com/Stackdriver/agent-packaging/blob/stackdriver-agent-5.5.2/deb/debian/rules#L37).
What `clean.sh` does is remove all of the autotools-generated scripts, so `build.sh` starts from a clean slate (in other words, it would replace the `autoreconf` call).

Reply:
I think there was a misunderstanding. I'm not trying to reconstruct our build environment, I'm only trying to run unit tests under ASAN. Can you see a good reason not to make this change? I think it's a strict improvement from the previous version.

I think it's great that you want to make this even better, but please do that as a separate change.

Reply:
You're pulling in bits from upstream that don't make sense for in our codebase. Invoking `build.sh` already essentially does for us what `autoreconf` does upstream.
So, `autoreconf` should not be required. If the change works fine without that and `clean.sh`, I'm fine with omitting `clean.sh` as well, but running both `autoreconf` and `build.sh` is what I'm mainly objecting to.

Reply:
Changed to include everything and to reflect what we run in the manual build.